src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EETypeNameDecoder.cs

```diff
@@ -20,9 +20,8 @@ internal sealed class EETypeNameDecoder : TypeNameDecoder<PEModuleSymbol, TypeSy
 
         protected override int GetIndexOfReferencedAssembly(AssemblyIdentity identity)
         {
-            var assemblies = this.GetAssemblies();
             // Find assembly matching identity.
-            int index = assemblies.IndexOf((assembly, id) => id.Equals(assembly.Identity), identity);
+            int index = this.Module.GetReferencedAssemblies().IndexOf(identity);
             if (index >= 0)
```

Comment:
Is this change necessary? It assumes that GetReferencedAssemblies and GetReferencedAssemblySymbols are aligned, which the previous code did not.